clusterous/cluster.py

```diff
@@ -643,6 +636,160 @@ def get_cluster_spec(self):
+            defaults.nat_ssh_port_forwarding, nat_public_ip, controller_private_ip))
+
+        ssh = paramiko.SSHClient()
+        ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+        ssh.connect(hostname = nat_public_ip,
+                    username = 'ec2-user',
+                    key_filename = os.path.expanduser(self._config['key_file']))
+        cmd = 'sudo iptables -t nat -A PREROUTING -p tcp --dport {0} -j DNAT --to-destination {1}:22'.format(
+             defaults.nat_ssh_port_forwarding, controller_private_ip)
+        # Using 'invoke_shell' because it is 'sudo' cmd
+        channel = ssh.invoke_shell()
+        channel.send('{0}\n'.format(cmd))
+        time.sleep(3)
+        output = channel.recv(99999)
+        ssh.close()
+        
+        if '000 -j DNAT' not in output:
+            message = 'Error: Setting SSH port forwarding on NAT instance.'
+            return (False, message)
+
+        return (True, '')
+
+    def _clusterous_tag(self, resource_name = None):
+        clusterous_tag = {defaults.instance_tag_key: self.cluster_name}
+        if resource_name is not None:
+            clusterous_tag.update({'Name': self.cluster_name + '-'+str(resource_name)})
+
+        return clusterous_tag
+    
+    def _create_vpc(self, vpc_conn, vpc_id = None):
+        vpc = None
+
+        if vpc_id is None:
+            vpc_name_full = "{0}-vpc".format(self.cluster_name)
+            
+            # Get list of VPCs
+            vpcs = vpc_conn.get_all_vpcs()
+            for i in vpcs:
+                if i.tags.get('Name','') == vpc_name_full:
+                    vpc = i
+        
+            if vpc is None:
+                # Create VPC
+                vpc = vpc_conn.create_vpc('10.2.0.0/16')
+                vpc.add_tags(self._clusterous_tag('vpc'))
+                vpc_conn.modify_vpc_attribute(vpc.id, enable_dns_support=True)
+                vpc_conn.modify_vpc_attribute(vpc.id, enable_dns_hostnames=True)
+        
+                # Create network acl
+                network_acl = vpc_conn.create_network_acl(vpc.id)
+                network_acl.add_tags(self._clusterous_tag('acl'))
+
+        else:
+            vpcs = vpc_conn.get_all_vpcs(vpc_ids=[vpc_id])
+            if len(vpcs) > 0:
+                vpc = vpcs[0]
+    
+        return vpc
+    
+    def _create_private_sg(self, vpc_conn, vpc, sg_name):
+        security_group = None
+        security_group_name_full = "{0}-{1}".format(self.cluster_name, sg_name)
+        security_groups = vpc_conn.get_all_security_groups()
+        for i in security_groups:
+            if i.tags.get('Name','') == security_group_name_full:
+                security_group = i
+    
+        if security_group is None:
+            # Private Security group
+            security_group = vpc_conn.create_security_group(self.cluster_name+"-{0}".format(sg_name), 'Private security group for ' + self.cluster_name, vpc.id)
```

Comment:
As per the other comment, it would be good if these names are generated based on a standard format string.